property-resolver-consul/consul.go

```diff
@@ -0,0 +1,57 @@
+package consulresolver
+
+import (
+	"github.com/hashicorp/consul/api"
+	"github.com/project-flogo/core/data/property"
+	"github.com/project-flogo/core/engine"
+	"github.com/project-flogo/core/support/log"
+	"strings"
+)
+
+var (
+	kv     *api.KV
+	logger = log.ChildLogger(log.RootLogger(), "consul-resolver")
+)
+
+func init() {
+	logger := log.RootLogger()
+
+	client, err := api.NewClient(api.DefaultConfig())
+	if err != nil {
+		logger.Error("Unable to initialize Consul property resolver for configuration management.")
+	}
+
+	kv = client.KV()
+
+	logger.Debug("Registering Consul resolver")
+
+	err = property.RegisterExternalResolver("consul", &SimpleConsulKVValueResolver{})
+
+	if err != nil {
+		logger.Error("Unable to register Consul property resolver for configuration management.")
+	}
+}
+
+// Resolve property value from a Consul KV Store
+type SimpleConsulKVValueResolver struct {
+}
+
+func (resolver *SimpleConsulKVValueResolver) LookupValue(key string) (interface{}, bool) {
+	key = strings.Replace(key, ".", "/", -1)
+
+	consul_key := "flogo/" + engine.GetAppName() + "/" + key
```

Comment:
Camel case naming ? _consulKey_

Reply:
Resolved by https://github.com/square-it/flogo-config-mgmt/pull/2/commits/b60b05acd9a1ff692520b8232dd6cdf88f880965